packages/common/src/audius-query/createApi.ts

```diff
@@ -553,6 +557,130 @@ const buildEndpointHooks = <
+      if (singlePageData) {
+        return state?.api?.[reducerPath]?.[endpointName]?.[
+          getKeyFromFetchArgs(args)
+        ]?.normalizedData
+      }
+
+      // By default, select data for every page up to the current page and aggregate the results
+      const pageAccumulator = []
+
+      for (let i = 0; i <= page; i++) {
+        const key = getKeyFromFetchArgs({
+          ...args,
+          offset: i * pageSize
+        })
+        const normalizedPageData =
+          state?.api?.[reducerPath]?.[endpointName]?.[key]?.normalizedData
+        if (normalizedPageData) {
+          pageAccumulator.push(...normalizedPageData)
+        }
+      }
+      return pageAccumulator
+    })
+
+    // Since this is a paginated query we know that the results will be in an array
+    // Call the useQuery hook to trigger cache checks and/or fetches
+    const result = useQuery(args, queryHookOptions) as QueryHookResults<Data[]>
+
+    // If forceLoad is set, force a data refresh on every render
+    useEffect(() => {
+      if (forceLoad) {
+        result.forceRefresh()
+        setForceLoad(false)
+      }
+    }, [result, forceLoad])
+
+    // Track status of the latest page fetches in order to update loadingMore
+    useCustomCompareEffect(
+      () => {
+        setStatus(result.status)
+        if (result.status === Status.ERROR) {
+          setLoadingMore(false)
+          return
+        }
+        if (result.status === Status.SUCCESS) {
+          setLoadingMore(false)
+        }
+      },
+      [result.status, args],
+      isEqual
+    )
+
+    const notError = result.status !== Status.ERROR
+    const stillLoadingCurrentPage =
+      loadingMore || result.status === Status.LOADING
+    const notStarted = result.status === Status.IDLE && allPageData.length === 0
+    const hasNotFetched = !result.data && result.status !== Status.SUCCESS
+    // If fetchedFullPreviousPage is less than the pageSize we can infer that there are no more pages to fetch
+    const fetchedFullPreviousPage = result.data?.length === pageSize
+
+    const hasMore =
+      notError && (notStarted || hasNotFetched || fetchedFullPreviousPage)
+
+    // Callback to increment the page state which will trigger new more data queries
+    const loadMore = useCallback(() => {
+      // If we're still loading a page OR there isn't any more data to load nothing should happen here yet
+      if (stillLoadingCurrentPage || !hasMore) {
+        return
+      }
+      setLoadingMore(true)
+      setPage(page + 1)
+    }, [stillLoadingCurrentPage, hasMore, page])
+
+    return {
+      ...result,
+      status:
+        // If not in singlePageData mode, we treat the status based on whether the first page succeeded. After that, use isLoadingMore for load statuses
+        allPageData?.length > 0 && !singlePageData ? Status.SUCCESS : status,
+      data: allPageData,
+      isLoadingMore: stillLoadingCurrentPage,
+      hardReset,
+      softReset,
```

Comment:
we could combine these into one `reset(hard?: boolean)`

Reply:
So true

Reply:
I like